systemtests/utils/vagrantnode.go

```diff
@@ -15,34 +15,114 @@ limitations under the License.
 
 package utils
 
+import (
+	"fmt"
+	"io/ioutil"
+
+	"golang.org/x/crypto/ssh"
+)
+
 // VagrantNode implements a node in vagrant testbed
 type VagrantNode struct {
-	Name    string
-	NodeNum int
+	Name   string
+	client *ssh.Client
+}
+
+//NewVagrantNode intializes a node in vagrant testbed
+func NewVagrantNode(name string, nodeNum int, port, privKeyFile string) (*VagrantNode, error) {
+	var (
+		vnode      *VagrantNode
+		err        error
+		signer     ssh.Signer
+		privateKey []byte
+	)
+
+	if privateKey, err = ioutil.ReadFile(privKeyFile); err != nil {
+		return nil, err
+	}
+
+	if signer, err = ssh.ParsePrivateKey(privateKey); err != nil {
+		return nil, err
+	}
+
+	config := &ssh.ClientConfig{
+		User: "vagrant",
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+	}
+
+	vnode = &VagrantNode{Name: name}
+	if vnode.client, err = ssh.Dial("tcp", fmt.Sprintf("127.0.0.1:%s", port), config); err != nil {
+		return nil, err
+	}
+
+	return vnode, nil
+}
+
+//Cleanup clears the ssh client resources
+func (n *VagrantNode) Cleanup() {
+	n.client.Close()
+}
+
+func newCmdStrWithSource(cmd string) string {
+	// we need to source the environment manually as the ssh package client
+	// doesn't do it automatically (I guess something to do with non interative
+	// mode)
+	return fmt.Sprintf("source /etc/profile.d/envvar.sh && %s", cmd)
 }
 
 // RunCommand runs a shell command in a vagrant node and returns it's exit status
-func (n VagrantNode) RunCommand(cmd string) error {
-	vCmd := &VagrantCommand{ContivNodes: n.NodeNum}
-	return vCmd.Run("ssh", n.Name, "-c", cmd)
+func (n *VagrantNode) RunCommand(cmd string) error {
+	var (
+		s   *ssh.Session
+		err error
+	)
+
+	if s, err = n.client.NewSession(); err != nil {
+		return err
+	}
+	defer s.Close()
```

Comment:
Yeah that would have been nice and I tried doing that first but it doesn't work. The ssh package supports only one command execution per session (https://godoc.org/golang.org/x/crypto/ssh#Session.CombinedOutput)

However, from the code the `session.Close()` itself didn't seem too expensive, they just seem to close a channel interface. Most of the work is done as part of `client.Dial()`